npm run build実行時に発生するエラーを解消し、npm run dev実行時にnpm run buildの内容を実行するように修正する

掲題の通り。

### やるべきこと
・`react-practice/package.json`の`scripts`の内容について、以下のように修正。
```json
  "scripts": {
    "dev": "npm run build && vite",
    "build": "npm run lint && tsc -b && vite build",
    "lint": "eslint .",
    "preview": "vite preview"
  },
```

・現状、`npm i`で依存パッケージを再インストールした後に`npm run lint`を実行すると以下のようなエラーが発生する。
下記のエラーが発生しないように実装する。
```bash
pbko-sato@3a9cb68a8a9c:~/react-practice [main]$ npm run build
> react-beginner-tutorial@0.0.0 build
> tsc -b && vite build

src/App_20250507.tsx:12:18 - error TS2739: Type '{ disabled: boolean; }' is missing the following properties from type 'ButtonProps': label, onClick
12                 <Button disabled={isChecked} />
                    ~~~~~~

src/Article.tsx:1:37 - error TS2307: Cannot find module 'react-router-dom' or its corresponding type declarations.
1 import { Outlet, useNavigate } from "react-router-dom";
                                      ~~~~~~~~~~~~~~~~~~
Found 2 errors.
```

--- a/package.json
+++ b/package.json
@@ -4,8 +4,9 @@
   "version": "0.0.0",
   "scripts": {
     "dev": "npm run build && vite",
-    "build": "npm run lint && tsc -b && vite build",
+    "build": "npm run lint && npm run type-check && vite build",
     "lint": "eslint .",
+    "type-check": "tsc -b",
     "preview": "vite preview"
   },
   "dependencies": {

--- a/package.json
+++ b/package.json
@@ -3,8 +3,8 @@
   "private": true,
   "version": "0.0.0",
   "scripts": {
-    "dev": "vite",
-    "build": "tsc -b && vite build",
+    "dev": "npm run build && vite",
+    "build": "npm run lint && tsc -b && vite build",
     "lint": "eslint .",
     "preview": "vite preview"
   },

--- a/src/App_20250507.tsx
+++ b/src/App_20250507.tsx
@@ -9,7 +9,9 @@ const App: React.FC = () => {
         <><div>
             <Checkbox isChecked={isChecked} onChange={setIsChecked} />
         </div><div>
-                <Button disabled={isChecked} />
+                <Button disabled={isChecked} label={''} onClick={function (): void {
+                    throw new Error('Function not implemented.');
+                }} />
             </div></>
     );
 }

--- a/src/Article.tsx
+++ b/src/Article.tsx
@@ -1,4 +1,4 @@
-import { Outlet, useNavigate } from "react-router-dom";
+import { Outlet, useNavigate } from "react-router";
 
 export const Article = () => {
     const navigate = useNavigate();

--- a/src/Posts.tsx
+++ b/src/Posts.tsx
@@ -29,7 +29,7 @@ export const Posts = () => {
     return (
         <div style={{ padding: '20px' }}>
             <h3>Here is posts!</h3>
-            {posts.map((post: any) => (
+            {posts.map((post: Post) => (
                 <div key={post.id}>
                     <h2
                         onClick={() => handleClick(post)}

--- a/src/components/ui/color-mode.tsx
+++ b/src/components/ui/color-mode.tsx
@@ -7,7 +7,7 @@ import type { ThemeProviderProps } from "next-themes"
 import * as React from "react"
 import { LuMoon, LuSun } from "react-icons/lu"
 
-export interface ColorModeProviderProps extends ThemeProviderProps {}
+export interface ColorModeProviderProps extends ThemeProviderProps { name: string }
 
 export function ColorModeProvider(props: ColorModeProviderProps) {
   return (
@@ -45,7 +45,7 @@ export function ColorModeIcon() {
   return colorMode === "dark" ? <LuMoon /> : <LuSun />
 }
 
-interface ColorModeButtonProps extends Omit<IconButtonProps, "aria-label"> {}
+interface ColorModeButtonProps extends Omit<IconButtonProps, "aria-label"> { name: string }
 
 export const ColorModeButton = React.forwardRef<
   HTMLButtonElement,

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -4,4 +4,5 @@ import react from '@vitejs/plugin-react'
 // https://vite.dev/config/
 export default defineConfig({
   plugins: [react()],
+  server: { host: true }
 })

